### PR TITLE
v4 entity api wrapper

### PR DESCRIPTION
**Pull request by duylk-1584**
## Related Tickets
- [#229036](https://dev.framgia.com/issues/229036)
- [#228575](https://dev.framgia.com/issues/228575)
- [#229037](https://dev.framgia.com/issues/229037)
- [#229030](https://dev.framgia.com/issues/229030)
- [#229028](https://dev.framgia.com/issues/229028)
- [#228578](https://dev.framgia.com/issues/228578)
- [#228575](https://dev.framgia.com/issues/228575)

## What's this PR do ?
- [x] Create entity 
- [x] Update entity
- [x] Delete entity
- [x] Get AWS upload key
- [x] Get status
- [x] Publish

## Performance
- [x] Resolved n + 1 query
- [x] Run explain query already

## Checklist
- [x] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [x] Note reason, scope of influence, solution into ticket
- [x] Validate UI/Model/API

### CODE REVIEW

File: lib/controllers/Entity.js [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
 const docResponse = require('../utils/docResponse').Entity;
 
 const generateRequestParams = require('../utils/generateRequestParams');
 
-exports.get = (data) => restClient().get(docResponse.get, `${pathURL}${generateRequestParams(data)}`);
+exports.retrieve = (data) => restClient().get(docResponse.get, `${pathURL}${generateRequestParams(data)}`);
+
+exports.list = () => restClient().get(docResponse.get, `${pathURL}${generateRequestParams()}`);
+
+exports.create = (data) => restClient().post(docResponse.create, `${pathURL}${generateRequestParams()}`, data);
```
Comment (NgoVanTien) [inline lib/controllers/Entity.js]: Bỏ generateRequestParams vẩn ok mà em nhĩ. Có cần pẩm trên url không ? Mấy cái kia tương tự

File: lib/controllers/Entity.js [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+
+exports.update = (data) => restClient().update(docResponse.update, `${pathURL}${generateRequestParams()}`, data);
+
+exports.publish = (id) => restClient().post(docResponse.publish, `${pathURL}/publish${generateRequestParams()}`, id);
```
Comment (NgoVanTien) [inline lib/controllers/Entity.js]: Cái này truyền vào là data chứ. Nó là object mà

**Comment by NgoVanTien** [comment]
Update thêm test và document nửa nhe

File: lib/controllers/Entity.js (hunk 2 above)
Comment (duylk-1584) [reply lib/controllers/Entity.js]: hiện nó đang là object id đó a :D

**Pull request merged**